### PR TITLE
New class constructors for DecFileParser

### PR DESCRIPTION
**Pull request by eduardo-rodrigues**
Addresses https://github.com/scikit-hep/decaylanguage/issues/50.

It also addresses trouble found by @dbobulsk parsing her user files.

### CODE REVIEW

**Comment by eduardo-rodrigues** [comment]
@henryiii, this is ready from my side.

File: decaylanguage/dec/dec.py [hunk 1]
```diff
@@ -87,33 +90,78 @@ def __init__(self, filename):
...
+        _cls._dec_file = stream.read()
 
-        # Check input file
-        if not os.path.exists(filename):
-            raise FileNotFoundError("'{0}'!".format(filename))
+        return _cls
 
-        return cls(filename)
+    @classmethod
+    def from_files(cls, filenames):
```
Comment (henryiii) [inline decaylanguage/dec/dec.py]: Couldn't either a) call the base constructor here with each file and merge, or b) allow a list of decay files in the base constructor?

File: decaylanguage/dec/dec.py [hunk 2]
```diff
@@ -65,17 +67,18 @@ class DecFileParser(object):
...
                  "_grammar_info",
-                 "_dec_file_name",
+                 "_dec_file_names",
+                 "_dec_file",
                  "_parsed_dec_file",
                  "_parsed_decays",
                  "_include_ccdecays")
 
-    def __init__(self, filename):
+    def __init__(self, filename=None):
```
Comment (henryiii) [inline decaylanguage/dec/dec.py]: This could be `self, *filenames`

Comment (henryiii) [reply decaylanguage/dec/dec.py]: The main constructor body would pretty much look just like the current from_files body.

**Comment by eduardo-rodrigues** [comment]
Yes, could be a good thing. I just need to see how to work that out together with the `from_string` class constructor, which also seems handy, hence worth keeping.

File: decaylanguage/dec/dec.py (hunk 1 above)
Comment (eduardo-rodrigues) [reply decaylanguage/dec/dec.py]: I've removed this class constructor making the default one versatile.

File: decaylanguage/dec/dec.py (hunk 2 above)
Comment (eduardo-rodrigues) [reply decaylanguage/dec/dec.py]: Yep, done.

**Comment by dbobulsk** [comment]
Many thanks for these new features which also address my issue, I'm gonna give it a try :)

**Pull request merged**